database.class.php

```diff
@@ -0,0 +1,235 @@
+                    echo "Successfull connection to database" . "\n";
+
+                    /* disable emulated prepared statements and use real prepared statements */
+                    $this->connection->setAttribute(PDO::ATTR_EMULATE_PREPARES, false);
+                    $this->connection->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+                } catch (PDOException $e) 
+                {
+                    die($e->getMessage());
+                }  
+            }
+            return $this->connection;    
+        }
+
+        /**
+         * Returns rows from the database based on the conditions
+         * @param string name of the table
+         * @param array select, where, order_by, limit and return_type conditions
+         */
+        public function selectData($table, $conditions = array())
+        {
+            $sql = 'SELECT ';
+            $sql .= array_key_exists("select",$conditions)?$conditions['select']:'*';
+            $sql .= ' FROM '.$table;
+
+            if(array_key_exists("where",$conditions))
+            {
+                $sql .= ' WHERE ';
+                $i = 0;
+
+                foreach($conditions['where'] as $key => $value)
+                {
+                    $pre = ($i > 0)?' AND ':'';
+                    $sql .= $pre.$key." = '".$value."'";
+                    $i++;
+                }
+            }
+
+            if(array_key_exists("order_by",$conditions))
+            {
+                $sql .= ' ORDER BY '.$conditions['order_by']; 
+            }
+
+            if(array_key_exists("start",$conditions) && array_key_exists("limit",$conditions))
+            {
+                $sql .= ' LIMIT '.$conditions['start'].','.$conditions['limit']; 
+
+            } elseif(!array_key_exists("start",$conditions) && array_key_exists("limit",$conditions))
+            {
+                $sql .= ' LIMIT '.$conditions['limit']; 
+            }
+
+            $query = $this->connection->prepare($sql);
+            $query->execute();
+
+            /**
+             * Initialize cache array with infos that we need for query caching
+             */
+            $cacheArray = $this->initCacheArray($sql);
+
+            /**
+             * Checks if we have cached data in folded "cache"
+             */
+            if ($this->isCached($cacheArray['cacheFile'], $cacheArray['cacheTimeSeconds'])) {
+                echo 'is cashed';
+
+                /* Store date */
+                $fileContents = file_get_contents($cacheArray['cacheFile']);
+
+                /* Decode the JSON back into an array. */
+                $data = json_decode($fileContents, true);
+
+            } else {
+                /* Execute SELECT query */
+                if(array_key_exists("return_type",$conditions) && $conditions['return_type'] != 'all')
```

Comment:
@lumakos It seems that `conditions` argument hide a lot of undocumented - therefore unexpected - functionality. Can you see a problem with this approach?

Reply:
SELECT COUNT(*) statement does not work as we want but im not sure, what do you mean?